### ISSUE TITLE
Document / implement issue triage process

### ISSUE THREAD

**Issue by gregdek** [ansible/community#36]
Need to go over our templates to make sure they say the right things.

**Comment by robynbergeron** [comment]
#action rbergeron to hack on workflow process for issue triage / future 
                   bot work. 
#action rbergeron to diagram Magical Things

**Comment by robynbergeron** [comment]
(more context in notes from meeting on 3/16)

**Comment by gregdek** [comment]
Will touch base on this on Wed 23 March.

**Comment by bob-smith** [comment]
Do we have a flow chart, I think that would be a good and clear way of documenting the process (and checking if it makes sense). Very easy for other people to look at and understand the different labels (via the bot or people)

**Comment by gregdek** [comment]
Not yet because it hasn't been defined, but I expect we will have one, since we also have one (outdated that needs to be updated) for the PR triage process.

**Comment by gregdek** [comment]
update: @robynbergeron is waiting on gliffy's online service to come back up :)

**Comment by gregdek** [comment]
ETA for first draft on friday.

**Comment by robynbergeron** [comment]
Okay! Here's some stuff.
 
A decent start to logic drafted here: https://github.com/robynbergeron/thingsandstuff/blob/master/issue_workflow_v1

And a first look at a visible workflow here: 
https://github.com/robynbergeron/thingsandstuff/blob/master/issue_workflow_draft1.png

**Comment by robynbergeron** [comment]
Open Questions after doing some digging and reading around:
1: We already use https://github.com/ansible/ansibot for auto-responding to issues. Do we want to build upon this? (Note: it is a private repo, why?)
2: Does ansibot run on a cron job or?
3: "We don't auto-close tickets." Do we want to change that?
4: Do we want to have differing behavior for core/extras repos than we have for ansible/ansible?
5: Do we want module maintainers to be able to add any priority flags to issues?6: Ansibot as it currently exists requests for people to edit the original issue. Do we want that to continue to be the behavior?
7: Is the label directly tied to boilerplate text, or some time to action, or both? Can something be labeled as pending_action but have a different boilerplate response depending on the circumstance?

**Comment by gregdek** [comment]
Looking to have closure on the spec by Wednesday 6 April so we can start implementing the week of 11 April, and maybe have the first bot done by 15 April.

**Comment by robynbergeron** [comment]
...and then we can pay our GitHub issue taxes? :)

**Comment by robynbergeron** [comment]
Okay, I've wrapped up writing the logic / boilerplate down and I think it's ready to move ahead. 

@gregdek -- please take a look -- drawing the chart is a bit more time consuming, so having a mini-sanity check before drawing the full picture would be nice

**Comment by gregdek** [comment]
Awesome -- a full spec! 

I think this is 99% there. Let's discuss the open questions at today's cwg meeting.

**Comment by robynbergeron** [comment]
Okay. Per comments in today's meeting --
12:35 <@gregdek> I guess what I'm proposing is that once we know we've got the initial data -- 
                 which should be entirely parseable from the template -- then we stop caring 
                 about the state and we assume that submitter and maintainer are working on it. 
                 Which is one blanket state "in progress" until it gets fixed, or the submitter 
                 says "nah I'm closing it" -- or the maintainer
12:35 <@gregdek> says "I ain't fixing it because it ain't a problem". We need to give them that 
                 option, I guess.
12:36 <@gregdek> And we keep pinging "in progress" issues to make sure people are still talking.

#action rbergeron to modify with simpler workflow

By thursday or friday of this week

**Comment by robynbergeron** [comment]
Additionally; Account for the wontfix/"not actually a problem" case.

**Comment by robynbergeron** [comment]
add some variances on "closed" (aka: notabug, wontfix, 
                   resolved by a PR, just gave som ehelp and resolved it, whatever)

**Comment by gregdek** [comment]
We'll be done with this by Friday the 13th (DUN DUN DUNNNNN!!!) and move into implementation of the bot.

**Comment by robynbergeron** [comment]
Okay, I simplified this, but idiotically did so without rereading the above details. 

So: https://github.com/robynbergeron/thingsandstuff/blob/master/issue_workflow_v2 for the moment. Going to do a v3 with just in_progress, but feeling like that may be ... not quite enough? Meh.

**Comment by robynbergeron** [comment]
Okay, simplified even more. 

I think this should work. 

https://github.com/robynbergeron/thingsandstuff/blob/master/issue_workflow_v3

Note that I have added a "same_issue" detail as part of the boilerplate -- a thing that additional folks can leave in a comment if they are experiencing the same issue. We could probably run a period report counting those same_issues -- or better yet, run a periodic report that coallates PRs as well as issues in a per-module  way -- this could allow folks to more easily triage if they see issues as well as PRs side by side with good descriptions.

**Comment by robynbergeron** [comment]
(No picture, sorry!)

**Comment by gregdek** [comment]
@resmo is implementing :)

**Comment by gregdek** [comment]
Gonna consider this piece done. Now on @resmo for initial implementation (thanks @resmo!)
